pages/1_orders_and_revenue.py

```diff
@@ -18,49 +18,163 @@
-# The framework has been set, but no visualization or data processing has been applied.
-# Please perform any data processing in this file and not within the filter files. We can discuss upon completion if it makes sense to add any code to the filters file.
-
-## KPI Metrics which need to be updated.
+# KPI Metrics
 with st.container():
     col1, col2, col3, col4 = st.columns(4)
 
+    total_revenue = data['total_amount'].sum()
+    number_of_orders = data['header_id'].nunique()
+    number_of_customers = data['customer_id'].nunique()
+    new_customers = data[data['customer_created_at'] == data['created_at']]['customer_id'].nunique()
+
     with col1:
-        st.metric(label="**Total Revenue**", value="TBD")
+        st.metric(label="**Total Revenue**", value=f"${total_revenue:,.2f}")
 
     with col2:
-        st.metric(label="**Number of Orders**", value="TBD")
+        st.metric(label="**Number of Orders**", value=number_of_orders)
 
     with col3:
-        st.metric(label="**Number of Customers**", value="TBD")
+        st.metric(label="**Number of Customers**", value=number_of_customers)
 
     with col4:
-        st.metric(label="**New Customers**", value="TBD")
+        st.metric(label="**New Customers**", value=new_customers)
+
+
+# Time series charts
+data['month'] = data['created_at'].dt.to_period('M').dt.to_timestamp()
 
-# Time series charts need to be built out
 with st.container():
     row1_col1, row1_col2 = st.columns(2)
     row2_col1, row2_col2 = st.columns(2)
 
     with row1_col1:
         st.markdown("**Total Revenue Over Time**")
+        revenue_over_time = data.groupby('month')['total_amount'].sum().reset_index()
+        fig = px.line(revenue_over_time, x='month', y='total_amount')
+        fig.update_yaxes(tickprefix='$', tickformat='~s', title_text='Revenue ($1000s)')
+        fig.update_xaxes(title_text='Month')
+        st.plotly_chart(fig, use_container_width=True)
 
     with row1_col2:
         st.markdown("**Total Orders Over Time**")
+        orders_over_time = data.groupby('month')['header_id'].nunique().reset_index()
+        fig = px.line(orders_over_time, x='month', y='header_id')
+        fig.update_yaxes(title_text='Orders')
+        fig.update_xaxes(title_text='Month')
+        st.plotly_chart(fig, use_container_width=True)
 
     with row2_col1:
         st.markdown("**Percent of Successful Payments Over Time**")
+        successful_payments = data[data['header_status'] == 'completed'].groupby('month')['payment_id'].nunique()
+        total_payments = data.groupby('month')['payment_id'].nunique()
+        percent_successful = (successful_payments / total_payments * 100).fillna(0).reset_index()
+        fig = px.line(percent_successful, x='month', y='payment_id')
+        fig.update_yaxes(range=[0, 100], title_text='Percentage (%)')
+        fig.update_xaxes(title_text='Month')
+        st.plotly_chart(fig, use_container_width=True)
```

Comment:
This chart really isn't showing us anything interesting. Since we removed the Successful payments over time KPI I would recommend we change this chart to show something different. Any ideas? I would just request we use this chart to show a graph that tells a story with this data.

Reply:
Yeah the logic I used was to count the unique payment_ids whenever status = completed. However, as in the sample data's case, the payment_id only exists if the payment is successful I think, which is why it becomes 100% always. But it may be the same case in some finance platforms based on the logic we wrote. So I agree on taking it out. 

Maybe we can swap to Average Order Size?

Reply:
@fivetran-reneeli I like where you are thinking. Maybe we do something to showcase the products being sold? I am thinking of a chart that helps a customer see which products are selling best over time. Or something along those lines. Thoughts?

Reply:
I see-- I think we can either do 
1. stacked bar chart of revenue over time that's broken up by product.
2. horizontal bar chart representing products, cumulative revenue for the given time period

I am leaning towards the latter since 1 is just pretty much the same as Revenue over Time which we have.

Reply:
I agree - Let's go with the latter!

Reply:
Updated1